test-with-coverage fails for example dir

example dirs that do not have *.go file in its root directory (where go.mod is present) then coverage fails with error
```
cover: cannot run go list: exit status 1
build .: cannot find module for path .

make: *** [test-with-coverage] Error 1
```

Exclude examples from coverage.

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,8 @@
 coverage.*
 
 /example/basic/basic
+example/grpc/client/client
+example/grpc/server/server
 /example/http/client/client
 /example/http/server/server
 /example/http-stackdriver/client/client

--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,7 @@ EXAMPLES := $(shell ./get_main_pkgs.sh ./example)
 ALL_DOCS := $(shell find . -name '*.md' -type f | sort)
 # All directories with go.mod files. Used in go mod tidy.
 ALL_GO_MOD_DIRS := $(shell find . -type f -name 'go.mod' -exec dirname {} \; | sort)
+ALL_COVERAGE_MOD_DIRS := $(shell find . -type f -name 'go.mod' -exec dirname {} \; | egrep -v '^./example' | sort)
 
 GOTEST_MIN = go test -v -timeout 30s
 GOTEST = $(GOTEST_MIN) -race
@@ -28,7 +29,7 @@ precommit: lint generate build examples test
 
 .PHONY: test-with-coverage
 test-with-coverage:
-	set -e; for dir in $(ALL_GO_MOD_DIRS); do \
+	set -e; for dir in $(ALL_COVERAGE_MOD_DIRS); do \
 	  echo "go test ./... + coverage in $${dir}"; \
 	  (cd "$${dir}" && \
 	    $(GOTEST_WITH_COVERAGE) ./... && \

